hivemq-edge/src/frontend/src/components/Mermaid.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect } from 'react'
+import mermaid from 'mermaid'
+import { Card, CardBody } from '@chakra-ui/react'
+
+export interface MermaidProps {
+  text: string
+}
+
+const DEFAULT_CONFIG = {
+  startOnLoad: true,
+  theme: 'base',
+  logLevel: 'fatal',
+  securityLevel: 'strict',
+  arrowMarkerAbsolute: false,
+  flowchart: {
+    htmlLabels: true,
+    curve: 'linear',
+  },
+  sequence: {
+    diagramMarginX: 50,
+    diagramMarginY: 10,
+    actorMargin: 50,
+    width: 150,
+    height: 65,
+    boxMargin: 10,
+    boxTextMargin: 5,
+    noteMargin: 10,
+    messageMargin: 35,
+    mirrorActors: true,
+    bottomMarginAdj: 1,
+    useMaxWidth: true,
+    rightAngles: false,
+    showSequenceNumbers: false,
+  },
+  gantt: {
+    titleTopMargin: 25,
+    barHeight: 20,
+    barGap: 4,
+    topPadding: 50,
+    leftPadding: 75,
+    gridLineStartPadding: 35,
+    fontSize: 11,
+    fontFamily: '"Open-Sans", "sans-serif"',
+    numberSectionStyles: 4,
+    axisFormat: '%Y-%m-%d',
+  },
+}
+
+export const Mermaid: React.FC<MermaidProps> = ({ text }) => {
+  useEffect(() => {
+    mermaid.initialize({ ...DEFAULT_CONFIG })
+  }, [])
+  useEffect(() => {
+    mermaid.contentLoaded()
+  }, [text])
+
+  return (
+    <Card>
+      <CardBody className="mermaid" sx={{ '& svg': { scale: '0.75 0.75' } }} data-processed={undefined}>
```

Comment:
Possibly indeed. Mermaid is a Vanilla JS with no good React wrapper, `classname` is the only thing that makes the connection between the React component and the Mermaid renderer. So I deal with the problem by making sure there is only one diagram at a time :-)
And no decision made on usefulness  of that visualisation so it might go at the next PR.